polarCoordinateTraversal.m

```diff
@@ -146,15 +147,16 @@
 radial_voxels = [current_voxel_ID_r];
 
 % III. TRAVERSAL PHASE
-tol = 10^-10;
 t = t_begin; 
+previous_transition_flag = false;
 while (t < t_end)
     % 1. Calculate tMaxR, tMaxTheta
-    [tMaxR, tStepR] = radial_hit(ray_origin, ray_direction, ...
-        current_voxel_ID_r, circle_center, circle_max_radius, delta_radius, r, t, verbose);
+    [tMaxR, tStepR, previous_transition_flag] = radial_hit(ray_origin, ray_direction, ...
+        current_voxel_ID_r, circle_center, circle_max_radius, delta_radius, t, previous_transition_flag, verbose);
+        if (current_voxel_ID_r + tStepR <= 0), return; end
     [tMaxTheta, tStepTheta] = angular_hit(ray_origin, ray_direction, current_voxel_ID_theta,...
-        num_angular_sections, circle_center, t, verbose);
-    
+        num_angular_sections, circle_center, t, false);
+    if (current_voxel_ID_r + tStepR == 0), return; end
```

Comment:
How come you have this check? The check should only occur after radial_hit if I'm not mistaken.